### PR TITLE
Assorted fixes

### PR DESCRIPTION
**Pull request by poettering**

### CODE REVIEW

File: src/basic/path-util.c [hunk 1]
```diff
@@ -772,6 +772,19 @@ bool hidden_or_backup_file(const char *filename) {
...
+         * with always new suffixes and that everybody else should just adjust to that, then it really should be on
+         * them. Hence, in future, let's not add any more entries. Instead, let's ask those packages to instead adopt
+         * one of the generic suffixes/prefixes for hidden files or backups, possibly augmented with an additional
+         * string. Specifically: there's now:
+         *
+         *    The generic suffixes "~" and ".bak" for backup files
+         *    The generic prefix "." for hidden files
+         *
+         * Thus, if a new package manager "foopkg" wants its own set of "foopkg-new", "foopkg-old", "foopkg-dist" or so
+         * registered, let's refuse that and ask them to use "foopkg-new.bak" or "foopkg-new~" instead.
```
Comment (keszybz) [inline src/basic/path-util.c:785]: Seems fair.

**Comment by keszybz** [comment]
```
======================================================================
FAIL: test_backup_file (__main__.SysvGeneratorTest)
init.d script with backup file
----------------------------------------------------------------------
Traceback (most recent call last):
  File "../test/sysv-generator-test.py", line 404, in test_backup_file
    ['foo.bak.service', 'foo.old.service', 'foo.service'])
AssertionError: Lists differ: ['foo.old.service', 'foo.service'] != ['foo.bak.service', 'foo.old.service', 'foo.service']
```

The test needs updating?

File: src/basic/path-util.c (hunk 1 above)
Comment (martinpitt) [inline src/basic/path-util.c:785]: Agreed, thanks. The rpm/dpkg/ucf stuff much predates concepts like `sysv-generator`, so I guess we have to grandfather them, but this is indeed a nuisance and too much guesswork.

**Comment by keszybz** [comment]
Please see additional proposed change in #3151.

**Comment by poettering** [comment]
Closing in favour of #3151.

**Pull request closed without merging**